src/redux/actions/samples/updateSampleFile.js

```diff
@@ -16,18 +17,22 @@ const updateSampleFile = (
 ) => async (dispatch, getState) => {
   const updatedAt = moment().toISOString();
   const sample = getState().samples[sampleUuid];
+  // we'll need to remove the hard-coded 10x tech type once we start
+  // supporting other types and save the chosen tech type in redux
+  const valid = checkIfFileValid(fileName, '10X Chromium');
 
   try {
     // Save sample only if upload is successful or error
     if (fileDiff.upload.status === UploadStatus.UPLOADED
       || fileDiff.upload.status === UploadStatus.UPLOAD_ERROR) {
       const diffObject = {
-        fileNames: sample.fileNames.add(fileName),
+        fileNames: sample.fileNames,
```

Comment:
why was this line changed? If filenames is still a set then it should not be a problem, right?

Reply:
Relates to the below comment

Reply:
Ohh right, so it is already added in the reducer which means this is unnecessary.

Looks good and I definitely think it is better to perform all the setup for updates in the reducer while the action just sends what it received (and/or performs api calls), however I think I would go further and remove the `fileNames` field from this diff.

Reply:
Also something else that we will need to do eventually is separate sampleFile updates from sampleFile posts to the api a little bit more, but that's for a different task.